lib/spack/spack/solver/concretize.lp

```diff
@@ -48,11 +32,21 @@ version_declared(Package, Version) :- version_declared(Package, Version, _).
 
 % If something is a package, it has only one version and that must be a
 % declared version.
-1 { version(Package, Version) : version_declared(Package, Version) } 1
- :- node(Package), error("Each node must have exactly one version").
-
-% A virtual package may have or not a version, but never has more than one
-:- virtual_node(Package), 2 { version(Package, _) }.
+{ version(Package, Version) : version_declared(Package, Version) }
+ :- node(Package).
+versions_conflict(Package, Version1, Version2)
+  :- node(Package),
+     version(Package, Version1),
+     version(Package, Version2),
+     Version1 < Version2.  % WLOG
+no_version(Package) :- node(Package), not version(Package, _).
+
+% A virtual package may or may not have a version, but never has more than one
+versions_conflict(Virtual, Version1, Version2)
+  :- virtual_node(Virtual),
+     version(Virtual, Version1),
+     version(Virtual, Version2),
+     Version1 < Version2. % WLOG
```

Comment:
make this comment more clear -- or, since it's all over, make it say, e.g., 

```
% see [1]
```

and have an explanation for `[1]` at the bottom of the file.